### ISSUE TITLE
Use Hausdorff Distance Loss Function for medical image segmentation

### ISSUE THREAD

**Issue by haibinswe** [Dale-Black/DistanceTransforms.jl#43]
Hello,

I recently came across your project and the associated paper "Optimized Hausdorff Distance Loss Function Based on a GPU Accelerated Distance Transform". I am interested in using the loss function proposed in your project with Python pytorch for medical image segmentation, and I would like to integrate it with the Dice loss function.

Could you provide some guidance on how to integrate the loss function from your project, which is written in Julia, with Python? Any help would be greatly appreciated.

Thank you!

**Comment by haibinswe** [comment]
Thank you for your reply. Have you tried juliacall to incorporate the loss into python code? I am sorry that I have little knowledge about julia, which made me hard to find out the losses written with julia. I am also excited that you are working on python example that is valuable and innovative.

**Comment by Dale-Black** [comment]
Not yet, but feel free to give it a shot while we work out some examples on our end. Below is a screenshot of the [README](https://github.com/cjdoris/PythonCall.jl)

<img width="835" alt="image" src="https://github.com/Dale-Black/DistanceTransforms.jl/assets/62632960/f0563b40-2ffa-460f-884d-8622f8befdcf">

**Comment by Dale-Black** [comment]
@haibinswe I put together a super simple google colab notebook that shows how to call this distance transform in Python

https://colab.research.google.com/drive/1-CDqQgrBHoxNqs2IbMebMRxsp0m21jSa?usp=sharing%5D

I will hopefully find time somewhat soon to show how to also use the loss function and even utilize the GPU via pytorch tensors. But if you have time you should be able to explore the PythonCall docs and that notebook which should get you most of the way there

**Comment by Dale-Black** [comment]
Okay that same notebook has now been updated to include an example of a pytorch tensor being passed into our distance transform function. I am going to now close this issue since that should be sufficient for now, although more work can be done

**Comment by nnagururu** [comment]
@Dale-Black  Hi, thanks so much for publishing this work. The colab notebook link seems to be broken right now. If you could tak a look that would be much appreciated.

**Comment by Dale-Black** [comment]
Sorry, that link was indeed broken. Does this work?

https://colab.research.google.com/drive/1moYjSiGEA4GoYD_8BeSw69fBPv2MtROZ?authuser=2

**Comment by nnagururu** [comment]
@Dale-Black getting the following error:

> Notebook loading error
> There was an error loading this notebook. Ensure that the file is accessible and try again.
> Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project.
> 
> https://drive.google.com/drive/?action=locate&id=1moYjSiGEA4GoYD_8BeSw69fBPv2MtROZ&authuser=2

**Comment by Dale-Black** [comment]
What about this one

https://colab.research.google.com/drive/1moYjSiGEA4GoYD_8BeSw69fBPv2MtROZ?usp=sharing

**Comment by nnagururu** [comment]
That works thank you! Also does the distance method, as implemented work for a 3d point set?

**Comment by Dale-Black** [comment]
Currently, vectors, matrices, and 3D arrays are all that are implemented. If you would like to add support for 3D point sets, I would be happy to review any PRs
